plugins/texteditor.koplugin/main.lua

```diff
@@ -179,6 +183,29 @@ Export text to QR code, that can be scanned, for example, by a phone.]]),
                     callback = function()
                         self.qr_code_export = not self.qr_code_export
                     end,
+                    separator = true,
+                },
+                {
+                    text = _("Clean text editor history"),
+                    enabled_func = function()
+                        return #self.history > 0
+                    end,
+                    keep_menu_open = true,
+                    callback = function(touchmenu_instance)
+                        UIManager:show(ConfirmBox:new{
+                            text = _("Clean text editor history?"),
+                            ok_text = _("Clean"),
+                            ok_callback = function()
+                                self.history = {}
+                                self.last_view_pos = {}
+                                -- remove history items from the parent menu
+                                for _ = #sub_item_table, 4, -1 do
+                                    table.remove(sub_item_table)
```

Comment:
I'd rather see you checking for the `_texteditor_id` attribute than assuming files stop at the 4th menu item (because the day we add some item in there, we'll surely forget to change this `4` to `5`, and as this feature will be rarely used, we'll get the first bug reports a year later :)

Reply:
Done